hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/ConnectionPool.java

```diff
@@ -210,24 +217,22 @@ public AtomicInteger getClientIndex() {
    * @return Connection context.
    */
   protected ConnectionContext getConnection() {
-
     this.lastActiveTime = Time.now();
-
-    // Get a connection from the pool following round-robin
-    ConnectionContext conn = null;
     List<ConnectionContext> tmpConnections = this.connections;
-    int size = tmpConnections.size();
-    // Inc and mask off sign bit, lookup index should be non-negative int
-    int threadIndex = this.clientIndex.getAndIncrement() & 0x7FFFFFFF;
-    for (int i=0; i<size; i++) {
-      int index = (threadIndex + i) % size;
-      conn = tmpConnections.get(index);
-      if (conn != null && conn.isUsable()) {
-        return conn;
+    for (ConnectionContext tmpConnection : tmpConnections) {
```

Comment:
Is there something more efficient we can do for this?
Like sorting the list when we make the connection usable?
It is not that expensive to do this but just in case.

Reply:
How about I create a new PR to fix this issue? And this PR is only for RouterRpcClient support using multiple sockets.

Reply:
I'm fine with that, this is just an optimization.